plugins/catalog/src/components/CatalogTable/CatalogTable.tsx

```diff
@@ -196,70 +199,95 @@ export const CatalogTable = (props: CatalogTableProps) => {
-        // presentation of refs in the table should in general be handled with
-        // EntityRefLink / EntityName components
-        name: humanizeEntityRef(entity, {
-          defaultKind: 'Component',
-        }),
-        entityRef: stringifyEntityRef(entity),
-        ownedByRelationsTitle: ownedByRelations
-          .map(r => humanizeEntityRef(r, { defaultKind: 'group' }))
-          .join(', '),
-        ownedByRelations,
-        partOfSystemRelationTitle: partOfSystemRelations
-          .map(r =>
-            humanizeEntityRef(r, {
-              defaultKind: 'system',
-            }),
-          )
-          .join(', '),
-        partOfSystemRelations,
-      },
-    };
-  });
-
-  const typeColumn = (columns || defaultColumns).find(c => c.title === 'Type');
-  if (typeColumn) {
-    typeColumn.hidden = !showTypeColumn;
-  }
-  const showPagination = rows.length > 20;
   const currentKind = filters.kind?.value || '';
   const currentType = filters.type?.value || '';
+  // TODO(timbonicus): remove the title from the CatalogTable once using EntitySearchBar
+  const titlePreamble = capitalize(filters.user?.value ?? 'all');
   const titleDisplay = [titlePreamble, currentType, pluralize(currentKind)]
     .filter(s => s)
     .join(' ');
 
+  const columns = props.columns || defaultColumns;
+  const title = `${titleDisplay} (${entities.length})`;
+  const actions = props.actions || defaultActions;
+  const options = {
+    actionsColumnIndex: -1,
+    loadingType: 'linear' as const,
+    showEmptyDataSourceMessage: !loading,
+    padding: 'dense' as const,
+    ...tableOptions,
+  };
+
+  if (enablePagination) {
+    return (
+      <PaginatedCatalogTable
+        columns={columns}
+        emptyContent={emptyContent}
+        isLoading={loading}
+        title={title}
+        actions={actions}
+        subtitle={subtitle}
+        options={options}
+        data={entities.map(toEntityRow)}
+        next={pageInfo.next}
+        prev={pageInfo.prev}
+      />
+    );
+  }
+
+  const rows = entities.sort(refCompare).map(toEntityRow);
+  const pageSize = 20;
+  const showPagination = rows.length > pageSize;
+
   return (
     <Table<CatalogTableRow>
       isLoading={loading}
-      columns={columns || defaultColumns}
+      columns={columns}
       options={{
         paging: showPagination,
-        pageSize: 20,
-        actionsColumnIndex: -1,
-        loadingType: 'linear',
-        showEmptyDataSourceMessage: !loading,
-        padding: 'dense',
+        pageSize: pageSize,
         pageSizeOptions: [20, 50, 100],
-        ...tableOptions,
+        ...options,
       }}
       title={`${titleDisplay} (${entities.length})`}
       data={rows}
-      actions={actions || defaultActions}
+      actions={actions}
       subtitle={subtitle}
       emptyContent={emptyContent}
     />
   );
 };
 
 CatalogTable.columns = columnFactories;
+
+function toEntityRow(entity: Entity) {
```

Comment:
can this be moved above first usage?

Reply:
since `toEntityRow ` is a secondary function I preferred to keep it below `CatalogTable`, which is the primary function of the file. Same goes in other files 😄

Reply:
Gotta say I generally prefer keeping the main component function at the bottom too, but we don't have strict guidelines around it :p